sphinxcontrib/redoc.py

```diff
@@ -30,10 +32,24 @@ def render(app):
         # relies on them.
         ctx.setdefault('opts', {})
 
+        # "Embed" mode: the JSON spec is embedded in the page
+        # TODO: only works for local files: add remote files download ?
+        if ctx.get('embed') is True:
+            # Parse & dump the spec to have it as properly formatted json
+            with io.open(os.path.join(app.confdir, ctx['spec'])) as specfp:
```

Comment:
Hm, I believe we should explicitly set an encoding. If I'm not mistaken by default will be used encoding extracted from current locale, but OpenAPI spec may use another encoding.

So I can think about two options:

* Explicitly use `UTF-8` and say that only `UTF-8` specs are supported. At least we will know for sure which encoding to use to deal with files no matter which system we're running on.

* Use encoding from Sphinx config: `env.config.source_encoding`, and say that the specs must have the same encoding as other Sphinx documents.

* Read encoding value from per-spec configuration.

While third item is more flexible, it would require additional changes and thorough thinking. I'd suggest to go with either (1) or (2) in this PR.

Reply:
I'll implement (1) for now. Since the spec is generated by something that has nothing to do with Sphinx, I'd be reluctant to tie them together.